src/lazy/text/matched.rs

```diff
@@ -286,15 +287,15 @@ pub(crate) enum MatchedString {
     /// The string is in multiple segments:
     ///     """hello,"""
     ///     """ world!"""
-    Long(MatchedLongString),
-}
-
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub(crate) struct MatchedLongString {
-    // TODO: Decide what (if anything) to store here.
-    //       Storing any collection of bytes or ranges means that this type cannot implement Copy,
-    //       which in turn means MatchedValue and EncodedTextValue also cannot implement Copy.
-    //       We probably also don't want to heap allocate just to match the long string.
+    Long,
```

Comment:
Just a thought—would it be useful/advantageous to have an `Empty` variant for an empty string?

Reply:
An empty string would be considered a `LongSingleSegmentWithoutEscapes`, yielding a reference to an empty slice of the input buffer. It should be close to optimal.